remount/remount.c

```diff
@@ -0,0 +1,605 @@
+/*
+ * Copyright (c) 2025 Pantacor Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <regex.h>
+#include <sys/mount.h>
+
+#include <jsmn/jsmnutil.h>
+
+#include "utils/fs.h"
+#include "utils/list.h"
+#include "utils/json.h"
+#include "utils/str.h"
+
+struct remount_entry {
+	char *exp;
+	char *opts;
+	struct dl_list list; // remount_entry
+};
+
+void parse_remount_entry(char *json, struct dl_list *remount_entries)
+{
+	printf("DEBUG: parsing remount entry '%s'...\n\r", json);
+
+	int tokc, n, ret = -1;
+	jsmntok_t *tokv;
+	jsmntok_t **k;
+
+	if (jsmnutil_parse_json(json, &tokv, &tokc) < 0) {
+		printf("WARN: JSON could not be parsed\n\r");
+		goto out;
+	}
+
+	k = jsmnutil_get_object_keys(json, tokv);
+	if (!k) {
+		printf("WARN: key not found\n\r");
+		goto out;
+	}
+	if (!(k + 1)) {
+		printf("WARN: value not found\n\r");
+		goto out;
+	}
+
+	struct remount_entry *p;
+	p = calloc(1, sizeof(struct remount_entry));
+	if (!p)
+		goto out;
+	dl_list_add_tail(remount_entries, &p->list);
```

Comment:
Maybe I'm wrong, but could be an error to add this to the list in this part of the code. Down below, I see a couple of `calloc()` calls not checked, and if they fail, probably you will end with a list with some invalid members?

Reply:
moved that to the end and added checks and freeup for calloc failures